### ISSUE TITLE
the output of the same input image is different

### ISSUE THREAD

**Issue by veryHope** [jusiro/FLAIR#4]

**Comment by askerlee** [comment]
Yeah the result is drastically different; it looks almost like random. How we are supposed to interpret the results?
![CS58509_R2](https://github.com/user-attachments/assets/7dd9d758-edb9-4430-87b1-6e3b58f9a07e)

For this image, 3 different runs:
Image-Text similarities:
[[ 2.438  0.951 -1.828 -1.884  2.602 -1.934  0.96   0.289 -0.282]]
Probabilities:
[[0.35  0.079 0.005 0.005 0.413 0.004 0.08  0.041 0.023]]

Image-Text similarities:
[[ 1.853 -0.314  0.856 -2.382  2.634  0.901  2.705  2.863  2.076]]
Probabilities:
[[0.096 0.011 0.035 0.001 0.21  0.037 0.225 0.264 0.12 ]]

Image-Text similarities:
[[ 4.129  4.457 -2.386 -2.121  2.172  2.341  0.292 -0.185  2.203]]
Probabilities:
[[0.347 0.482 0.001 0.001 0.049 0.058 0.007 0.005 0.051]]

The textual labels are:
["normal", "healthy", "macular edema", "diabetic retinopathy", "glaucoma", "macular hole", "lesion", "lesion in the macula", "myopia"]

**Comment by jusiro** [comment]
Hello @askerlee,

Sorry for the late response.

I have reproduced your example with the image you provided. Specifically, I have executed the following code:

```
from PIL import Image
import numpy as np
# Import FLAIR
from flair import FLAIRModel
# Set model
model = FLAIRModel(from_checkpoint=True)
# Load image and set target categories 
# (if the repo is not cloned, download the image and change the path!)
image = np.array(Image.open("./documents/example_fundus_github.png"))
text = ["normal", "healthy", "macular edema", "diabetic retinopathy", "glaucoma", "macular hole",
        "lesion", "lesion in the macula", "myopia"]
# First run
print("First forward:")
probs, logits = model(image, text)
print("Image-Text similarities:")
print(logits.round(3)) 
print("Probabilities:")
print(probs.round(3)) 
# Second run
probs, logits = model(image, text)
print("Second forward:")
print("Image-Text similarities:")
print(logits.round(3)) 
print("Probabilities:")
print(probs.round(3)) 
```

The output is the following:

```
First forward:
Image-Text similarities:
[[-1.6    1.181  1.063  0.646  1.16  10.389  7.185  6.544 12.785]]
Probabilities:
[[0.    0.    0.    0.    0.    0.083 0.003 0.002 0.912]]
Second forward:
Image-Text similarities:
[[-1.6    1.181  1.063  0.646  1.16  10.389  7.185  6.544 12.785]]
Probabilities:
[[0.    0.    0.    0.    0.    0.083 0.003 0.002 0.912]]
```

As you can appreciate, the output of FLAIR given the same input remains constant. You might want to check if the model weights have been downloaded properly, and ensure that the model is still in "evaluation" mode, this is, model.eval(). Otherwise, I would need more details to reproduce your issue.

Kind regards,
